npb: fix mpi rank mismatch errors

viperML opened the pull request:
MPI variant has several rank mismatch errors, which are silently ignored. This downgrades the errors to warnings.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

viperML commented:
Pinging @adamjstewart because of https://github.com/spack/spack/pull/3762

becker33 commented on var/spack/repos/builtin/packages/npb/package.py:
```diff
@@ -129,7 +129,7 @@ def edit(self, spec, prefix):
                 "FMPI_LIB": spec["mpi"].libs.ld_flags,
                 "FMPI_INC": "-I" + spec["mpi"].prefix.include,
-                "FFLAGS": "-O3",
+                "FFLAGS": "-O3 -fallow-argument-mismatch",
```
Is that for all compilers or specific to some compilers/compiler versions?

viperML replied:
@becker33 I added a constrain on the GCC version that changes this rule, I don't know if this is the "idiomatic" way to do it

viperML committed:
npb: fix mpi rank mismatch errors

MPI variant has several rank mismatch errors, which are silently
ignored. This downgrades the errors to warnings.
